device/device.go

```diff
@@ -147,423 +31,28 @@ func Init(typeFlag string) (err error) {
-// the name of vbox doesn't change
-func selectVboxInit(conf string, v []vbox.Config) int {
-	opts := []string{
-		"Use default vbox preset",
-		"Create a new vbox preset",
-		"Use saved vbox preset",
+		fmt.Println("[-] Error: ", err)
+		return
 	}
-	optTypes := []int{
-		VBoxTypeDefault,
-		VBoxTypeNew,
-		VBoxTypeUser,
+	if err := df.PrepareForFlashing(); err != nil {
+		fmt.Println("[-] Error: ", err)
+		return
 	}
-	n := len(opts)
-
-	if _, err := os.Stat(conf); os.IsNotExist(err) || v == nil {
-		n--
+	if err := df.Configure(); err != nil {
+		fmt.Println("[-] Error: ", err)
+		return
 	}
-
-	return optTypes[dialogs.SelectOneDialog("Please select an option: ", opts[:n])]
 }
 
-// Starts a vbox, inits repository, downloads the image into repository, then uploads and unpacks it into the vbox
-func vboxDownloadImage(wg *sync.WaitGroup, vBoxTemplate, deviceType string) (*virtualbox.Machine, workstation.WorkStation, *vbox.Config, string) {
-	w := workstation.NewWorkStation()
-	help.ExitOnError(vbox.CheckDeps("VBoxManage"))
-
-	conf := filepath.Join(help.UserHomeDir(), ".iotit", "virtualbox", "iotit-vbox.json")
-	v := vbox.NewConfig(vBoxTemplate, deviceType)
-	vm, name, description, err := setVbox(v, conf, vBoxTemplate, deviceType)
-	help.ExitOnError(err)
-
-	if vm.State != virtualbox.Running {
-		fmt.Printf("[+] Selected virtual machine \n\t[\x1b[34mName\x1b[0m] - \x1b[34m%s\x1b[0m\n\t[\x1b[34mDescription\x1b[0m] - \x1b[34m%s\x1b[0m\n",
-			name, description)
-		progress := make(chan bool)
-		wg.Add(1)
-		go func(progress chan bool) {
-			defer close(progress)
-			defer wg.Done()
-
-			err := vm.Start()
-			help.ExitOnError(err)
-			time.Sleep(20 * time.Second)
-		}(progress)
-
-		help.WaitAndSpin("starting", progress)
-		wg.Wait()
-	}
-
-	repository, err := repo.NewRepository(deviceType)
-	help.ExitOnError(err)
-	dst := filepath.Join(repository.Dir(), repository.GetVersion())
-
-	fmt.Println("[+] Starting download ", deviceType)
-	zipName, bar, err := repo.DownloadAsync(repository, wg)
-	help.ExitOnError(err)
-
-	bar.Prefix(fmt.Sprintf("[+] Download %-15s", zipName))
-	bar.Start()
-	wg.Wait()
-	bar.Finish()
-	time.Sleep(time.Second * 2)
-
-	err = deleteHost(filepath.Join((os.Getenv("HOME")), ".ssh", "known_hosts"), "localhost")
-	if err != nil {
-		log.Error(err)
-	}
-
-	// 4. upload edison img
-	fmt.Printf("[+] Uploading %s to virtual machine\n", zipName)
-	err = v.SCP(filepath.Join(dst, zipName), constants.TMP_DIR)
-	help.ExitOnError(err)
-
-	// 5. unzip edison img (in VM)
-	fmt.Printf("[+] Extracting %s \n", zipName)
-	log.Debug("Extracting an image")
-	out, err := v.RunOverSSHExtendedPeriod(fmt.Sprintf("unzip %s -d %s", filepath.Join(constants.TMP_DIR, zipName), constants.TMP_DIR))
-	help.ExitOnError(err)
-
-	log.Debug(out)
-
-	str := strings.Split(zipName, ".")
-
-	return vm, w, v, strings.Join(str[:len(str)-1], ".") + ".img"
+// Prints sd card flashed message
+func printDoneMessageSd(device, username, password string) {
```

Comment:
__[gometalinter]__
* [deadcode] printDoneMessageSd is unused

<!-- Comment by [SideCI](https://sideci.com) -->